### PR TITLE
[TargetDefinition] Add swift_version to target_definition

### PR DESCRIPTION
**Pull request by DanToml**
Required for https://github.com/CocoaPods/CocoaPods/pull/5540

### CODE REVIEW

**Comment by segiddins** [comment]
👍 with test coverage

**Comment by DanToml** [comment]
@segiddins 👍, done!

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 1]
```diff
@@ -362,6 +362,26 @@ def uses_frameworks?
...
+      # @param  [Bool] version
+      #         The Swift version that the target definition should use.
+      #
+      # @return [void]
+      #
+      def swift_version=(version)
+        set_hash_value('swift_version', version)
+      end
+
+      # @return [Bool] whether the target definition should build
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: this comment looks copy-pasted

Comment (DanToml) [inline lib/cocoapods-core/podfile/target_definition.rb]: Oops. Will fix in a mo

**Pull request merged**